File: apps/web/src/api/hello.ts
import http from "./http";

export interface HelloResponse {
  status: string;
  data: {
    greeting: string;
  };
}

export async function sayHello(name: string): Promise<HelloResponse> {
  const { data } = await http.get<HelloResponse>("/api/v1/hello", {
    params: { name },
  });
  return data;
}

File: apps/api/app/api/v1/endpoints/hello.py
from fastapi import APIRouter, Query

router = APIRouter()


@router.get("/hello", summary="Say hello")
def say_hello(name: str = Query("world", description="Name to greet")) -> dict:
    greeting = f"Hello, {name}!"
    return {"status": "ok", "data": {"greeting": greeting}}
